core/src/main/java/org/eclipse/krazo/core/ViewResponseFilter.java

```diff
@@ -1,246 +1,285 @@
+import javax.ws.rs.Priorities;
+import javax.ws.rs.Produces;
+import javax.ws.rs.ServerErrorException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.core.Variant;
+
+import org.eclipse.krazo.KrazoConfig;
+import org.eclipse.krazo.engine.Viewable;
+import org.eclipse.krazo.event.ControllerRedirectEventImpl;
+import org.eclipse.krazo.lifecycle.RequestLifecycle;
+
+/**
+ * <p>
+ * A JAX-RS response filter that fires a {@link javax.mvc.event.AfterControllerEvent} event. It also verifies the static
+ * return type of the controller method is correct, and ensures that the entity is a {@link Viewable} to be processed by
+ * {@link ViewableWriter}.
+ * </p>
+ *
+ * <p>
+ * The class uses {@link javax.ws.rs.core.Request} which implements the algorithm in Section 3.8 of the JAX-RS
+ * specification to compute the final Content-Type when the method returns void (no entity). If unable to compute the
+ * final Content-Type, e.g. if the controller method is not annotated by {@code @Produces}, it defaults to
+ * {@code text/html}. If the method does not return void (has an entity), the computation of the Content-Type is done by
+ * JAX-RS and is available via {@code responseContext}.
+ * </p>
+ *
+ * <p>
+ * Given that this filter is annotated with {@link Controller}, it will be called after every controller method returns.
+ * Priority is set to {@link javax.ws.rs.Priorities#ENTITY_CODER} which means it will be executed after user-defined
+ * response filters (response filters are sorted in reverse order).
+ * </p>
+ *
+ * @author Santiago Pericas-Geertsen
+ */
+@Controller
+@Priority(Priorities.ENTITY_CODER)
+public class ViewResponseFilter implements ContainerResponseFilter {
+
+    private static final Logger LOGGER = Logger.getLogger(ViewResponseFilter.class.getName());
+
+    public static final String FILTER_EXECUTED_KEY = ViewResponseFilter.class.getName() + ".EXECUTED";
+
+    public static final String REDIRECT = "redirect:";
+
+    @Context
+    private UriInfo uriInfo;
+
+    @Context
+    private ResourceInfo resourceInfo;
+
+    @Context
+    private HttpServletRequest request;
+
+    private Event<MvcEvent> dispatcher;
+
+    private Messages messages;
+
+    private KrazoConfig krazoConfig;
+
+    private RequestLifecycle requestLifecycle;
+
+    private ViewPathResolver viewPathResolver;
+
+    @Deprecated
+    public ViewResponseFilter() {
+        // empty contructor for cdi eyes
+    }
+
+    @Inject
+    public ViewResponseFilter(Event<MvcEvent> dispatcher, Messages messages, KrazoConfig krazoConfig, RequestLifecycle requestLifecycle,

```

Comment:
Although I like the idea of using constructor injection here, let's not change too much in this PR. That makes it really hard to review.